text/0000-user-ids.md

```diff
@@ -0,0 +1,42 @@
+# Secure runtime environments
+
+## Summary
+
+This RFC recommends that Paketo Buildpacks should in general strive to avoid making run-time modifications to the Buildpacks layers or the application directory by default. Instead, they should consider the output image to be read-only by default and explicitly set the parts that need to be writeable during runtime.
+
+
+## Motivation
+
+As of Buildpacks [RFC 0085](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md) the Cloud Native Buildpacks project recommends that the run-time and build-time user ids be different. The motivation behind this RFC was to improve the security of output images and prevent arbitrary runtime modifications to built artifacts.
+
+The run-time and build-time user can potentially share the same group id if they wish to make modifications to the output image during run-time, by changing the permissions of the files they wish to modify during run-time to be group writable during the build process. Ideally most buildpacks should use the temporary directories or other known scratch directories for ephemeral runtime files.
+
+## Detailed Explanation
+
+This RFC recommends that we identify buildpacks that making runtime modifications to the application directory or buildpack layers and modify them to instead use a temporary directory - or have them make these files group writable instead. All the current Paketo stacks' build-time and run-time users belong to a unique group so changing the file permissions to be group writable should have no additional security implications for users using the Paketo stacks.
+
+
+## Rationale and Alternatives
+
+
+The rationale behind this RFC is explained in the upstream RFC - [RFC 0085](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md)
+
+In the future we could also modify the Paketo stacks to have different run-time/build-time user ids once all the buildpacks follow this practice for a more secure experience.
```

Comment:
No strong feelings one way or the other, but I'd be curious to know what folks think about doing this as part of this RFC? Why wait and have a second RFC? 

It kind of seems like it would be helpful to coordinate so that we're not making stack changes that break buildpacks.

Reply:
We'd need to think through the impacts this would have on the `watchexec` features we provide. Most of this is relying on seeing updates to files in the `/workspace`.

Reply:
It's also worth thinking about whether there are common app setups that rely on modifying the app's working directory. One example is the .NET app in [this issue](https://github.com/paketo-buildpacks/dotnet-execute/issues/258).
If we made this change to stacks, I think it'd be necessary to provide end-users with a mechanism to reach in and mark some files as group writeable. That way, users have a way around Paketo's security opinion on this.

Perhaps, we create a buildpack that accepts a user-defined set of files through an environment variable and marks all of them group-writeable.